pyls/workspace.py

```diff
@@ -24,19 +23,28 @@ class Workspace(object):
     M_SHOW_MESSAGE = 'window/showMessage'
 
     def __init__(self, root_uri, lang_server=None):
-        self._url_parsed = urlparse(root_uri)
-        self.root = unquote(self._url_parsed.path)
+        self._root_uri = root_uri
+        self._root_uri_scheme = uris.urlparse(self._root_uri)[0]
+        self._root_path = uris.to_fs_path(self._root_uri)
```

Comment:
I wonder if there is a way to avoid calling urlparse twice here. Even though it might not be a big deal it is a waste of resources, calling `uris.urlparse()` to convert the whole path and throw away everything but the scheme and then calling `uris.urlparse()` again in `uris.to_fs_path()`.